### PR TITLE
Add Chunk.Utilization() methods

### PR DESCRIPTION
**Pull request by juliusv**
When using the chunking code in other projects (both Weave Prism and
ChronixDB ingester), you sometimes want to know how well you are
utilizing your chunks when closing/storing them.

### CODE REVIEW

**Comment by juliusv** [comment]
@beorn7

File: storage/local/chunk/varbit.go [hunk 1]
```diff
@@ -322,6 +322,11 @@ func (c varbitChunk) UnmarshalFromBuf(buf []byte) error {
 // encoding implements chunk.
 func (c varbitChunk) Encoding() Encoding { return Varbit }
 
+// Utilization implements chunk.
+func (c varbitChunk) Utilization() float64 {
+	return float64(c.nextSampleOffset()) / float64(cap(c))
```
Comment (beorn7) [inline storage/local/chunk/varbit.go]: You might want to count the footer part at the end of the chunk as used, too. Not sure about the exact math, but with the current implementation, your utilization can never reach 100% (which might or might not be what you want).

**Comment by beorn7** [comment]
Just a math nit. Decide yourself what you need for your use case. Otherwise :+1:

**Comment by juliusv** [comment]
Updated varbit utilization function together with @beorn7.

**Comment by juliusv** [comment]
:+1: from @beorn7 in person

**Pull request merged**